lib/index.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+const yaml = require('js-yaml');
+const Joi = require('joi');
+const { serviceSchema } = require('joi-docker-compose');
+
+/**
+ * This library is responsible for generating an engine spec compatible with Docker Engine v1.36
+ * from a docker-compose formatted YAML file (version 3).
+ */
+
+function isNumeric(value) {
+  return !isNaN(parseFloat(value)) && isFinite(value);
+}
+
+function dockerComposeToApi(input) {
+  if (typeof input === "string") {
+    input = yaml.safeLoad(input);
+  }
+
+  let validation = Joi.validate(input, serviceSchema);
+  if (validation.error) {
+    console.log(validation.error);
```

Comment:
See asana and https://github.com/firstandthird/docker-compose-to-api/pull/1 for comments and example doc.